api/src/main/java/io/minio/PutOptions.java

```diff
@@ -0,0 +1,77 @@
+/*
+  * Minio Java SDK for Amazon S3 Compatible Cloud Storage, (C) 2018 Minio, Inc.
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *     http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+ 
+package io.minio;
+
+import java.util.Map;
+
+/**
+* PUT object options for parameterizing a PUT request.
+*/
+public class PutOptions extends Options {
```

Comment:
This class is not necessary. We don't have a Options-concept in minio-java (unfortunately) and adding it is out-of-scope of this PR. It requires design before adding it.